### PR TITLE
Add concurrency with test array in matrix

### PR DESCRIPTION
**Pull request by dnestoro**

### CODE REVIEW

File: .github/workflows/test-native-gradle-plugin.yml [hunk 1]
```diff
@@ -65,7 +83,7 @@ jobs:
           java-version: ${{ matrix.java-version }}
           github-token: ${{ secrets.GITHUB_TOKEN }}
       - name: "❓ Check and test the plugin"
-        run: ./gradlew :native-gradle-plugin:functionalTest -DgradleVersion=${{ matrix.gradle-version }}
+        run: ./gradlew :native-gradle-plugin:functionalTest -DgradleVersion=${{ matrix.gradle-version }} --tests ${{ matrix.test }}
       - name: "❓ Check and test the plugin with configuration cache"
         run: ./gradlew :native-gradle-plugin:configCacheFunctionalTest -DgradleVersion=${{ matrix.gradle-config-cache-version }}
```
Comment (fniephaus) [inline .github/workflows/test-native-gradle-plugin.yml:88]:
```suggestion
        run: ./gradlew :native-gradle-plugin:configCacheFunctionalTest -DgradleVersion=${{ matrix.gradle-config-cache-version }} --tests ${{ matrix.test }}
```

File: .github/workflows/test-native-gradle-plugin.yml [hunk 2]
```diff
@@ -65,9 +80,9 @@ jobs:
...
           github-token: ${{ secrets.GITHUB_TOKEN }}
       - name: "❓ Check and test the plugin"
-        run: ./gradlew :native-gradle-plugin:functionalTest -DgradleVersion=${{ matrix.gradle-version }}
+        run: ./gradlew :native-gradle-plugin:functionalTest -DgradleVersion=${{ matrix.gradle-version }} --tests ${{ matrix.test }}
       - name: "❓ Check and test the plugin with configuration cache"
-        run: ./gradlew :native-gradle-plugin:configCacheFunctionalTest -DgradleVersion=${{ matrix.gradle-config-cache-version }}
+        run: ./gradlew :native-gradle-plugin:configCacheFunctionalTest -DgradleVersion=${{ matrix.gradle-config-cache-version }}  --tests ${{ matrix.test }}
       - name: "📜 Upload functional tests results"
         if: always()
         uses: actions/upload-artifact@v2
```
Comment (fniephaus) [inline .github/workflows/test-native-gradle-plugin.yml:88]: Please bump all uses of `actions/upload-artifact` to `v3` the next time you push:
```suggestion
        uses: actions/upload-artifact@v3
```

File: .github/workflows/test-native-gradle-plugin.yml [hunk 3]
```diff
@@ -22,9 +22,30 @@ concurrency:
...
+      - name: "🕸️ Populate matrix"
+        id: set-matrix
+        run: |
+          ./gradlew -PmatrixType=gradle :native-gradle-plugin:dumpFunctionalTestList
+
   test-native-gradle-plugin:
     name: "Sanity checks"
-    runs-on: ${{ matrix.os }}
+    runs-on: "ubuntu-20.04"
+    timeout-minutes: 60
```
Comment (fniephaus) [inline .github/workflows/test-native-gradle-plugin.yml:48]: where does this timeout come from? One native test shouldn't take longer than maybe 15min, no? Can we use a timeout of 20 or 30min maybe?

File: .github/workflows/test-native-gradle-plugin.yml [hunk 4]
```diff
@@ -22,9 +22,30 @@ concurrency:
...
+      - name: "🔧 Prepare environment"
+        uses: graalvm/setup-graalvm@v1
+        with:
+          java-version: '17'
+          distribution: 'graalvm'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: "🕸️ Populate matrix"
+        id: set-matrix
+        run: |
+          ./gradlew -PmatrixType=gradle :native-gradle-plugin:dumpFunctionalTestList
```
Comment (fniephaus) [inline .github/workflows/test-native-gradle-plugin.yml:43]:
```suggestion
        run: ./gradlew -PmatrixType=gradle :native-gradle-plugin:dumpFunctionalTestList
```

File: .github/workflows/test-native-gradle-plugin.yml [hunk 5]
```diff
@@ -78,18 +100,15 @@ jobs:
         with:
           name: functional-tests-results-graalvm-dev
           path: native-gradle-plugin/build/reports/tests/functionalTest/
+
   functional-testing-gradle-plugin:
-    name: "Functional testing"
+    name: "🧪 Gradle: ${{ matrix.test }} on ${{ matrix.os }} with gradle version: ${{ matrix.gradle-version }}"
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
```
Comment (fniephaus) [inline .github/workflows/test-native-gradle-plugin.yml:107]: Some here, can this be reduced to 20/30min?

File: .github/workflows/test-native-gradle-plugin.yml [hunk 6]
```diff
@@ -98,37 +117,54 @@ jobs:
...
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: "🕸️ Populate matrix"
+        id: set-matrix
+        run: |
+          ./gradlew -PmatrixType=gradleCached :native-gradle-plugin:dumpFunctionalTestList
+
+  functional-testing-gradle-with-cache-version:
+    name: "🧪 Gradle: ${{ matrix.test }} on ${{ matrix.os }} with cache gradle version: ${{ matrix.gradle-config-cache-version }}"
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
```
Comment (fniephaus) [inline .github/workflows/test-native-gradle-plugin.yml:151]: And again

File: .github/workflows/test-native-maven-plugin.yml [hunk 7]
```diff
@@ -22,51 +22,46 @@ concurrency:
...
-    # Fails with "Illegal char <�> at index 3: C:\�Users\�RUNNER~1\�AppData\�Local\�Temp\�spock_supports_spaces_in__0_testDirectory18198684407774777428\�with spaces\�local-repo\�org\�apache\�commons\�commons-lang3\�3.8.1\�commons-lang3-3.8.1.jar"
+      - name: "🕸️ Populate matrix"
+        id: set-matrix
+        run: |
+          ./gradlew -PmatrixType=maven :native-maven-plugin:dumpFunctionalTestList
+
+  test-native-maven-plugin:
+    name: "🧪 Maven: ${{ matrix.test }} on ${{ matrix.os }}"
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
```
Comment (fniephaus) [inline .github/workflows/test-native-maven-plugin.yml:48]: Last time

**Review by fniephaus — CHANGES_REQUESTED** [review]

File: .github/workflows/test-native-gradle-plugin.yml (hunk 3 above)
Comment (dnestoro) [reply .github/workflows/test-native-gradle-plugin.yml:48]: Now that we have almost final overview how much time jobs take, we can decrease timeout. I checked jobs and the longest one took 30min, so probably 40min should be okay limit. What do you think?

**Review by fniephaus — APPROVED** [review]

**Pull request merged**